src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -39,29 +39,19 @@ public static partial class Math
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static short Abs(short value)
         {
-            if (value < 0)
+            unchecked
             {
-                value = (short)-value;
-                if (value < 0)
-                {
-                    ThrowAbsOverflow();
-                }
+                return (short)((value + 0xFFFF) ^ 0xFFFF);
```

Comment:
Unfortunately this is a breaking change.

Previously, `Math.Abs(short.MinValue)` would throw:
```
Negating the minimum value of a twos complement number is invalid.
  + System.Math.AbsHelper(short)
```

and now, it returns `short.MinValue`, which is invalid for `Abs`.

Reply:
It might be interesting to see the perf if this was changed to:
```
if (value == short.MinValue)
{
    ThrowAbsOverflow();
}

return (short)((value + 0xFFFF) ^ 0xFFFF);
```

which would reduce this to a single branch, rather than two. However, this might also hurt codegen in some cases if the JIT can determine that `value` is definitely not negative.

Either way, it would be good to see some benchmarks.

Reply:
Thanks Tanner, good catch!

Unfortunately putting the if in to check makes it slower than the original.

```
unchecked
            {
                short result = (short) ((value + 0xFFFF) ^ 0xFFFF);
                if (result < 0) ThrowAbsOverflow();               
                return result;
            }
```

Reply:
What if you put the check first (`if value == short.MinValue`)? Ideally that would be slightly more efficient since it is fewer checks in a small code window, and would be predicted false by default.

Reply:
Sorry Tanner, I didn't realise you'd replied.

I benchmarked by running each for 3,276,700 iterations in Release mode (10 times to avoid JIT and drive out statistical anomalies). But I also only ran it with predictable negative numbers (-32767 thu 0), for the "unhappy" path.

Removed unchecked { ... } because my project's set to unchecked.

This is very slightly slower than the original:

```
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public static short ShortAbs(short value)
        {
            if (value == short.MinValue) ThrowAbsOverflow();
            short result = value<0? (short) ((value + ffff) ^ ffff):value;            
            return result;
        }
```

And some of the timings, i7-4940MX. Using Stopwatch, Elapsed.TotalMilliseconds and Elapsed.Ticks/3276700f*100 (100ns per tick)

```
New: 1.8733ms total, 0.5717033ns per invocation
Old: 1.7169ms total, 0.5239723ns per invocation
New: 1.8574ms total, 0.5668508ns per invocation
Old: 1.7329ms total, 0.5288553ns per invocation
New: 1.8803ms total, 0.5738395ns per invocation
Old: 1.7653ms total, 0.5387433ns per invocation
New: 1.8562ms total, 0.5664846ns per invocation
Old: 1.7409ms total, 0.5312967ns per invocation
```

And the following code gets similar call times, interestingly:

```
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public static short ShortAbs(short value)
        {
            if (value == short.MinValue) ThrowAbsOverflow();
            return value < 0 ? (short)-value : value;            
        }
```